Java/core/src/main/java/com/amazon/randomcutforest/store/IPointStore.java

```diff
@@ -27,12 +27,26 @@
 
     P get(int index);
 
+    void incrementRefCount(int index);
+
+    void decrementRefCount(int index);
+
+    /**
+     * Adds to the store; there may be a loss of precision if enableFloat is on in
+     * the Forest level. But external interface of the forest is double[]
+     * 
+     * @param point point to be added
+     * @return index of the stored point
+     */
+    int add(double[] point);
```

Comment:
Previously this interface exposed a read-only interface, which I thought was good because it meant that trees couldn't change the point store (only the update coordinator could do that). If we still think that's useful, we could create an IReadOnlyPointStore (or something) and add the mutating methods to a sub-interface.

Reply:
Great suggestion. Done. I used IPointStoreReadOnly instead of IReadOnlyPointStore -- the latter seems to be like a statement.

Reply:
The other nice thing about this naming convention is that IPointStore and IPointStoreReadOnly will appear next to each other when listed alphabetically.

Reply:
Yes, as we specialize implementation, it'll be good to keep library internal assumptions separate from use.